signals_windows.go

```
@@ -25,39 +25,136 @@ import (
+	"RTMIN+12": linuxSigrtmin + 12,
+	"RTMIN+13": linuxSigrtmin + 13,
+	"RTMIN+14": linuxSigrtmin + 14,
+	"RTMIN+15": linuxSigrtmin + 15,
+	"RTMAX-14": linuxSigrtmax - 14,
+	"RTMAX-13": linuxSigrtmax - 13,
+	"RTMAX-12": linuxSigrtmax - 12,
+	"RTMAX-11": linuxSigrtmax - 11,
+	"RTMAX-10": linuxSigrtmax - 10,
+	"RTMAX-9":  linuxSigrtmax - 9,
+	"RTMAX-8":  linuxSigrtmax - 8,
+	"RTMAX-7":  linuxSigrtmax - 7,
+	"RTMAX-6":  linuxSigrtmax - 6,
+	"RTMAX-5":  linuxSigrtmax - 5,
+	"RTMAX-4":  linuxSigrtmax - 4,
+	"RTMAX-3":  linuxSigrtmax - 3,
+	"RTMAX-2":  linuxSigrtmax - 2,
+	"RTMAX-1":  linuxSigrtmax - 1,
+	"RTMAX":    linuxSigrtmax,
 }
 
 // ParseSignal parses a given string into a syscall.Signal
 // the rawSignal can be a string with "SIG" prefix,
 // or a signal number in string format.
 func ParseSignal(rawSignal string) (syscall.Signal, error) {
+	return parseSignalGeneric(rawSignal, "windows")
+}
+
+// ParsePlatformSignal parses a given string into a syscall.Signal based on
+// the OS platform specified in `platform`.
+func ParsePlatformSignal(rawSignal, platform string) (syscall.Signal, error) {
+	return parseSignalGeneric(rawSignal, platform)
+}
+
+func parseSignalGeneric(rawSignal, platform string) (syscall.Signal, error) {
+	signalMap := getSignalMapForPlatform(platform)
 	s, err := strconv.Atoi(rawSignal)
 	if err == nil {
 		sig := syscall.Signal(s)
+		if platform != "windows" {
+			return sig, nil
+		}
+		// on windows, make sure we support this signal
 		for _, msig := range signalMap {
 			if sig == msig {
 				return sig, nil
 			}
 		}
-		return -1, fmt.Errorf("unknown signal %q", rawSignal)
+		return sig, fmt.Errorf("unknown signal %q", rawSignal)
 	}
 	signal, ok := signalMap[strings.TrimPrefix(strings.ToUpper(rawSignal), "SIG")]
 	if !ok {
 		return -1, fmt.Errorf("unknown signal %q", rawSignal)
 	}
 	return signal, nil
 }
+
+func getSignalMapForPlatform(platform string) map[string]syscall.Signal {
+	if platform != "windows" {
```

I think it would be better to return an error if we are passed an unrecognized platform. Otherwise we could end up with a bug in the future where the linux signal map is used unintentionally.